app/src/main/java/mozilla/lockbox/view/RootActivity.kt

```diff
@@ -51,4 +58,24 @@ class RootActivity : AppCompatActivity() {
             super.onBackPressed()
         }
     }
-}
+
+    fun showToastNotification(action: ToastNotificationAction) {
+        assertOnUiThread()
+
+        val toast = Toast(this)
+        toast.duration = Toast.LENGTH_SHORT
+        val container = window.decorView.rootView as ViewGroup
+        toast.view = layoutInflater.inflate(R.layout.toast_view, container, false)
+        toast.setGravity(Gravity.FILL_HORIZONTAL or Gravity.BOTTOM, 0, 0)
+
+        // set text
+        val view = toast.view.findViewById(R.id.message) as TextView
+        view.text = action.viewModel.text ?: resources.getString(action.viewModel.strId!!)
```

Comment:
I think you should change the `viewModel` here to force us to use a `strId`, with an optional `text` argument.

The we have: 

```kotlin
            // set text
            val vm = action.viewModel
            val message = toast.view.findViewById(R.id.message) as TextView
            message.text = vm.text?.let {
                resources.getString(vm.strId, it)
            } ?: resources.getString(vm.strId)
```

Reply:
Note that we use `strId` in each case, but we insert the untranslatable`vm.text` in to the translatable string resource.

Reply:
Ah, gotcha. This _doesn't_ compile.